### PR TITLE
Added docstyle to pylint, and preformed the suggested refactor.

### PR DESCRIPTION
**Pull request by thanacles**

### CODE REVIEW

**Comment by thanacles** [comment]
This is related to Issue: https://github.com/quantumlib/Cirq/issues/3388

add to docstyle to pylint and performed the suggested refactor.

File: cirq-aqt/cirq_aqt/aqt_sampler.py [hunk 1]
```diff
@@ -41,10 +41,9 @@ class AQTSampler(cirq.Sampler):
     """
 
     def __init__(self, remote_host: str, access_token: str):
-        """
-        Args:
-            remote_host: Address of the remote device.
-            access_token: Access token for the remote api.
+        """Args:
```
Comment (balopat) [inline cirq-aqt/cirq_aqt/aqt_sampler.py:44]: First line should be a description of the method. In this case I think the args should be moved to `Attributes` on the class. https://google.github.io/styleguide/pyguide.html#384-classes

File: cirq-core/cirq/circuits/optimization_pass.py [hunk 2]
```diff
@@ -36,22 +36,21 @@ def __init__(
...
-            new_operations: The operations to replace the cleared out
-                operations with.
-            preserve_moments: If set, `cirq.Moment` instances within
-                `new_operations` will be preserved exactly. Normally the
-                operations would be repacked to fit better into the
-                target space, which may move them between moments.
-                Please be advised that a PointOptimizer consuming this
-                summary will flatten operations no matter what,
-                see https://github.com/quantumlib/Cirq/issues/2406.
+        """Args:
```
Comment (balopat) [inline cirq-core/cirq/circuits/optimization_pass.py:39]: First line should be a description of the method. In this case I think the args should be moved to `Attributes` on the class. https://google.github.io/styleguide/pyguide.html#384-classes

File: cirq-core/cirq/circuits/optimization_pass.py [hunk 3]
```diff
@@ -92,11 +91,10 @@ def __init__(
             [Sequence['cirq.Operation']], ops.OP_TREE
         ] = lambda op_list: op_list,
     ) -> None:
-        """
-        Args:
-            post_clean_up: This function is called on each set of optimized
-                operations before they are put into the circuit to replace the
-                old operations.
+        """Args:
```
Comment (balopat) [inline cirq-core/cirq/circuits/optimization_pass.py:94]: First line should be a description of the method. In this case I think the args should be moved to `Attributes` on the class. https://google.github.io/styleguide/pyguide.html#384-classes

File: cirq-core/cirq/contrib/graph_device/graph_device.py [hunk 4]
```diff
@@ -125,15 +125,13 @@ def __init__(
         device_graph: Optional[UndirectedHypergraph] = None,
         crosstalk_graph: Optional[UndirectedHypergraph] = None,
     ) -> None:
```
Comment (balopat) [inline cirq-core/cirq/contrib/graph_device/graph_device.py:127]: First line should be a description of the method. In this case I think the args should be moved to `Attributes` on the class. https://google.github.io/styleguide/pyguide.html#384-classes

**Review by balopat — CHANGES_REQUESTED** [review]
Thanks for doing this! I think this is looking promising - one major issue at first blink: can you change the constructors to follow the Google style guide? (I know, we are all over the place regarding to it in the code base). 

Namely, the first line should be always a short description of the method, we should never start with `Args`. In the constructor case the args should be moved to `Attributes` on the class. https://google.github.io/styleguide/pyguide.html#384-classes

I'll keep reviewing.

**Comment by thanacles** [comment]
@balopat That makes sense, for most of the cases.  I'll rename Args -> Attributes and move the comment from the constructor to the class, but it does seem like a few of the constructor args are being assigned to private variables.  I feel we shouldn't mark those as attributes.

**Comment by thanacles** [comment]
How about this,  I'll update the constructor doc with "Inits ClassName" and leave arguments that get assigned to private variables there.

**Comment by balopat** [comment]
> How about this, I'll update the constructor doc with "Inits ClassName" and leave arguments that get assigned to private variables there.

I think you're right - `Attributes` should be only used for classes that are just value types, e.g. dataclasses. And it would be a bit of a code smell to have public attributes on classes that define some kind of behavior based on initialization parameters.

Sounds good. Let's keep them separate, add a one liner for the constructors.

**Comment by thanacles** [comment]
@balopat Done.  What do you think?

**Pull request closed without merging**